tests/pc/test_po_cleanup.py

```diff
@@ -27,36 +27,38 @@ def ignore_expected_loganalyzer_exceptions(rand_one_dut_hostname, loganalyzer):
         ]
         loganalyzer[rand_one_dut_hostname].ignore_regex.extend(ignoreRegex)
         expectRegex = [
-            ".*teamd#teammgrd: :- cleanTeamProcesses.*",
-            ".*teamd#teamsyncd: :- cleanTeamSync.*"
+            ".*teammgrd: :- cleanTeamProcesses.*",
+            ".*teamsyncd: :- cleanTeamSync.*"
         ]
         loganalyzer[rand_one_dut_hostname].expect_regex.extend(expectRegex)
 
-
-def check_kernel_po_interface_cleaned(duthost):
-    res = duthost.shell("ip link show | grep -c PortChannel",  module_ignore_errors=True)["stdout_lines"][0].decode("utf-8")
+def check_kernel_po_interface_cleaned(duthost, asic_index):
+    namespace = duthost.get_namespace_from_asic_id(asic_index)
+    res = duthost.shell(duthost.get_linux_ip_cmd_for_namespace("ip link show | grep -c PortChannel", namespace),module_ignore_errors=True)["stdout_lines"][0].decode("utf-8")
     return res == '0'
 
-
-def test_po_cleanup(duthosts, rand_one_dut_hostname, tbinfo):
-    """
-    test port channel are cleaned up correctly and teammgrd and teamsyncd process
-    handle  SIGTERM gracefully
-    """
+@pytest.fixture(scope="module", autouse=True)
+def check_topo_and_restore(duthosts, rand_one_dut_hostname, tbinfo):
+    
     duthost = duthosts[rand_one_dut_hostname]
     mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
 
     if len(mg_facts['minigraph_portchannels'].keys()) == 0:
```

Comment:
We would need to check this per namespace/asic_index  now  ? as the test_po_cleanup is run per asic

Reply:
@judyjoseph updated. 
Will still check only once and not per asic and for multi-asic it will always run as backend is always port-channel.